woocommerce/woocommerce-ios
# Dynamic Dashboard: Optimize API calls

Closes: #12770 


## Description


Stops sending unnecessary network calls for dashboard cards that are not enabled. 

Changes
- For store performance, top performance and coupon cards - Stop sending reload calls when the view model is initialized. 
- DashboardViewModel
    - Add a separate method to sync essential data upon view load.
    - Reload cards only if the card is enabled.
    - Reload only newly enabled card. 
## Testing instructions


- Fresh install the app.
- Login into the app.
- If you have orders in your store, you should see the store performance and top performers cards.
- If your store is eligible for Blaze, the Blaze card should be displayed in the dashboard. 
- If you have Proxyman or any other setup to interrupt network calls, test that network requests are sent only for enabled cards. 
- Tap "Edit" and enable any other card. 
- Save and ensure that the card is added to the dashboard and the content is loaded.
- Repeat the above steps for all the cards and validate that each card is loaded as expected.
- Close the app. 
- Relaunch and ensure that the previous configuration of cards is loaded in the dashboard.
- If you have Proxyman or any other setup to interrupt network calls, test that network requests are sent only for enabled cards. 

## Screenshots

| Dashboard cards | Customize screen |
|--------|--------|
| ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-18 at 18 57 22](https://github.com/woocommerce/woocommerce-ios/assets/524475/f810389b-4b99-49b8-8a10-6c3cfb236ae2) | ![Simulator Screenshot - iPhone 15 Pro Max - 2024-05-18 at 18 57 24](https://github.com/woocommerce/woocommerce-ios/assets/524475/efd110b1-ebf2-42a4-b913-6ebb0989894c) | 

---
